src/main.c

```diff
@@ -117,14 +118,21 @@ int main(int argc, char *argv[])
             break;
         case 'm':
             mode = strtol(optarg, &endp, 10) - 1;
+            // mode = desired communication mode from terminal input - 1
             if (optarg == endp
                 || MODE_MAX_VALUE < mode
                 || MODE_MIN_VALUE > mode)
             {
                 fprintf(stderr, "Invalid mode\n");
                 printf("%s", usage);
+                // For debug
+                //printf("Communication mode passed to rl78 funcs: %i\r\n", mode);
+                //printf("Mode value expression: %i\r\n",mode_val_expr);
                 return EINVAL;
             }
+            // For debug
+            //printf("Communication mode passed to rl78 funcs: %i\r\n", mode);
+            //printf("Mode value expression: %i\r\n",mode_val_expr);
```

Comment:
Normally, you'd remove your temporary debugging code from the commit before issuing a PR.  Or, wrap it in a `-d` CLI parameter.